### PR TITLE
8309332: RISC-V: Improve PrintOptoAssembly output of vector nodes

### PR DESCRIPTION
**Pull request by zifeihan**
Hi, Currently in the vector node implementation, some of the instruction formats are part of the rvv assembly instructions, for which the assembly instructions are also incomplete, such as the missing vsetvli assembly instruction(vsetvli is used to set the element width, number, etc), This makes the compiler assembly output by -XX:+PrintOptoAssembly difficult to understand. And if every assembly instruction is reflected, it will be redundant and increase the maintenance work in the future. And referring to other CPUs, such as the ARM64, it is straightforward to use the instruct function name to simplify [1].

While this won't affect release build, we should fix this for debug build. We can use the -XX:+PrintOptoAssembly parameter to print the compiler assembly output by -XX:+PrintOptoAssembly and view the assembly logic in conjunction with the source code of the specific vector node.

[1] https://github.com/openjdk/jdk/blob/4460429d7a50b9a7a99058ef4e5ae36fb30b956f/src/hotspot/cpu/aarch64/aarch64_vector.ad#L2842-L2846

### Testing:
qemu with UseRVV:

- [ ] Tier1 tests (release)
- [x] test/jdk/jdk/incubator/vector (fastdebug)
- [x] test/jdk/jdk/incubator/vector/Int256VectorTests.java (fastdebug with -XX:+PrintOptoAssembly)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8309332](https://bugs.openjdk.org/browse/JDK-8309332): RISC-V: Improve PrintOptoAssembly output of vector nodes

### Reviewers
 * [Yanhong Zhu](https://openjdk.org/census#yzhu) (@yhzhu20 - Author) ⚠️ Review applies to [9e0430c0](https://git.openjdk.org/jdk/pull/14279/files/9e0430c0120c167d89d76f6d20a4ca468ad626df)
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14279/head:pull/14279` \
`$ git checkout pull/14279`

Update a local copy of the PR: \
`$ git checkout pull/14279` \
`$ git pull https://git.openjdk.org/jdk.git pull/14279/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14279`

View PR using the GUI difftool: \
`$ git pr show -t 14279`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14279.diff">https://git.openjdk.org/jdk/pull/14279.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14279#issuecomment-1573039669)

### CODE REVIEW

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 1]
```diff
@@ -1072,7 +1071,7 @@ instruct vneg(vReg dst, vReg src) %{
   match(Set dst (NegVI src));
   match(Set dst (NegVL src));
   ins_cost(VEC_COST);
-  format %{ "vrsub.vx $dst, $src, $src\t#@vneg" %}
+  format %{ "vneg $dst, $src, $src" %}
```
Comment (yhzhu20) [inline src/hotspot/cpu/riscv/riscv_v.ad:1074]: Hi, for vneg and vfneg, I think `vneg $dst, $src` is better.

Comment (zifeihan) [reply src/hotspot/cpu/riscv/riscv_v.ad]: Hi, Thanks for your review. and has been fixed.

**Review by yhzhu20 — APPROVED** [review]

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 2]
```diff
@@ -342,7 +341,7 @@ instruct vadd_masked(vReg dst_src1, vReg src2, vRegMask_V0 v0) %{
   match(Set dst_src1 (AddVI (Binary dst_src1 src2) v0));
   match(Set dst_src1 (AddVL (Binary dst_src1 src2) v0));
   ins_cost(VEC_COST);
-  format %{ "vadd.vv $dst_src1, $src2, $v0\t#@vadd_masked" %}
+  format %{ "vadd_masked $dst_src1, $src2, $v0" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:344]: Suggestion: `format %{ "vadd_masked $dst_src1, $dst_src1, $src2, $v0" %}`

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 3]
```diff
@@ -357,7 +356,7 @@ instruct vadd_fp_masked(vReg dst_src1, vReg src2, vRegMask_V0 v0) %{
   match(Set dst_src1 (AddVF (Binary dst_src1 src2) v0));
   match(Set dst_src1 (AddVD (Binary dst_src1 src2) v0));
   ins_cost(VEC_COST);
-  format %{ "vfadd.vv $dst_src1, $src2, $v0\t#@vadd_fp_masked" %}
+  format %{ "vadd_fp_masked $dst_src1, $src2, $v0" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:359]: Suggestion: `format %{ "vadd_fp_masked $dst_src1, $dst_src1, $src2, $v0" %}`

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 4]
```diff
@@ -408,7 +407,7 @@ instruct vsub_masked(vReg dst_src1, vReg src2, vRegMask_V0 v0) %{
   match(Set dst_src1 (SubVI (Binary dst_src1 src2) v0));
   match(Set dst_src1 (SubVL (Binary dst_src1 src2) v0));
   ins_cost(VEC_COST);
-  format %{ "vsub.vv $dst_src1, $src2, $v0\t#@vsub_masked" %}
+  format %{ "vsub_masked $dst_src1, $src2, $v0" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:410]: Suggestion: `format %{ "vsub_masked $dst_src1, $dst_src1, $src2, $v0" %}`

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 5]
```diff
@@ -790,7 +789,7 @@ instruct vfmlsF(vReg dst_src1, vReg src2, vReg src3) %{
   match(Set dst_src1 (FmaVF dst_src1 (Binary (NegVF src2) src3)));
   match(Set dst_src1 (FmaVF dst_src1 (Binary src2 (NegVF src3))));
   ins_cost(VEC_COST);
-  format %{ "vfnmsac.vv $dst_src1, $src2, $src3\t#@vfmlsF" %}
+  format %{ "vfmlsF $dst_src1, $src2, $src3" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:792]: Suggestion: `format %{ "vfmlsF $dst_src1, $dst_src1, $src2, $src3" %}`

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 6]
```diff
@@ -806,7 +805,7 @@ instruct vfmlsD(vReg dst_src1, vReg src2, vReg src3) %{
   match(Set dst_src1 (FmaVD dst_src1 (Binary (NegVD src2) src3)));
   match(Set dst_src1 (FmaVD dst_src1 (Binary src2 (NegVD src3))));
   ins_cost(VEC_COST);
-  format %{ "vfnmsac.vv $dst_src1, $src2, $src3\t#@vfmlsD" %}
+  format %{ "vfmlsD $dst_src1, $src2, $src3" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:808]: Similar here.

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 7]
```diff
@@ -841,7 +840,7 @@ instruct vfnmlaF(vReg dst_src1, vReg src2, vReg src3) %{
   match(Set dst_src1 (FmaVF (NegVF dst_src1) (Binary (NegVF src2) src3)));
   match(Set dst_src1 (FmaVF (NegVF dst_src1) (Binary src2 (NegVF src3))));
   ins_cost(VEC_COST);
-  format %{ "vfnmacc.vv $dst_src1, $src2, $src3\t#@vfnmlaF" %}
+  format %{ "vfnmlaF $dst_src1, $src2, $src3" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:843]: Similar here.

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 8]
```diff
@@ -857,7 +856,7 @@ instruct vfnmlaD(vReg dst_src1, vReg src2, vReg src3) %{
   match(Set dst_src1 (FmaVD (NegVD dst_src1) (Binary (NegVD src2) src3)));
   match(Set dst_src1 (FmaVD (NegVD dst_src1) (Binary src2 (NegVD src3))));
   ins_cost(VEC_COST);
-  format %{ "vfnmacc.vv $dst_src1, $src2, $src3\t#@vfnmlaD" %}
+  format %{ "vfnmlaD $dst_src1, $src2, $src3" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:859]: Similar here.

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 9]
```diff
@@ -890,7 +889,7 @@ instruct vfnmlsF(vReg dst_src1, vReg src2, vReg src3) %{
   predicate(UseFMA);
   match(Set dst_src1 (FmaVF (NegVF dst_src1) (Binary src2 src3)));
   ins_cost(VEC_COST);
-  format %{ "vfmsac.vv $dst_src1, $src2, $src3\t#@vfnmlsF" %}
+  format %{ "vfnmlsF $dst_src1, $src2, $src3" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:892]: Similar here.

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 10]
```diff
@@ -904,7 +903,7 @@ instruct vfnmlsD(vReg dst_src1, vReg src2, vReg src3) %{
   predicate(UseFMA);
   match(Set dst_src1 (FmaVD (NegVD dst_src1) (Binary src2 src3)));
   ins_cost(VEC_COST);
-  format %{ "vfmsac.vv $dst_src1, $src2, $src3\t#@vfnmlsD" %}
+  format %{ "vfnmlsD $dst_src1, $src2, $src3" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:906]: Similar here.

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 11]
```diff
@@ -939,7 +938,7 @@ instruct vmla(vReg dst_src1, vReg src2, vReg src3) %{
   match(Set dst_src1 (AddVI dst_src1 (MulVI src2 src3)));
   match(Set dst_src1 (AddVL dst_src1 (MulVL src2 src3)));
   ins_cost(VEC_COST);
-  format %{ "vmacc.vv $dst_src1, src2, src3\t#@vmla" %}
+  format %{ "vmla $dst_src1, src2, src3" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:941]: Similar here.

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 12]
```diff
@@ -975,7 +974,7 @@ instruct vmls(vReg dst_src1, vReg src2, vReg src3) %{
   match(Set dst_src1 (SubVI dst_src1 (MulVI src2 src3)));
   match(Set dst_src1 (SubVL dst_src1 (MulVL src2 src3)));
   ins_cost(VEC_COST);
-  format %{ "vnmsac.vv $dst_src1, src2, src3\t#@vmls" %}
+  format %{ "vmls $dst_src1, src2, src3" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:977]: Similar here.

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 13]
```diff
@@ -1042,7 +1041,7 @@ instruct vmul_masked(vReg dst_src1, vReg src2, vRegMask_V0 v0) %{
   match(Set dst_src1 (MulVI (Binary dst_src1 src2) v0));
   match(Set dst_src1 (MulVL (Binary dst_src1 src2) v0));
   ins_cost(VEC_COST);
-  format %{ "vmul.vv $dst_src1, $src2, $v0\t#@vmul_masked" %}
+  format %{ "vmul_masked $dst_src1, $src2, $v0" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:1044]: Similar here.

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 14]
```diff
@@ -1056,7 +1055,7 @@ instruct vmul_fp_masked(vReg dst_src1, vReg src2, vRegMask_V0 v0) %{
   match(Set dst_src1 (MulVF (Binary dst_src1 src2) v0));
   match(Set dst_src1 (MulVD (Binary dst_src1 src2) v0));
   ins_cost(VEC_COST);
-  format %{ "vmul.vv $dst_src1, $src2, $v0\t#@vmul_fp_masked" %}
+  format %{ "vmul_fp_masked $dst_src1, $src2, $v0" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:1058]: Similar here.

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 15]
```diff
@@ -1392,9 +1375,7 @@ instruct reduce_addF(fRegF src1_dst, vReg src2, vReg tmp) %{
   match(Set src1_dst (AddReductionVF src1_dst src2));
   effect(TEMP tmp);
   ins_cost(VEC_COST);
-  format %{ "vfmv.s.f $tmp, $src1_dst\t#@reduce_addF\n\t"
-            "vfredosum.vs $tmp, $src2, $tmp\n\t"
-            "vfmv.f.s $src1_dst, $tmp" %}
+  format %{ "reduce_addF $src1_dst, $src2\t# KILL $tmp" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:1378]: Similar here.

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 16]
```diff
@@ -1409,9 +1390,7 @@ instruct reduce_addD(fRegD src1_dst, vReg src2, vReg tmp) %{
   match(Set src1_dst (AddReductionVD src1_dst src2));
   effect(TEMP tmp);
   ins_cost(VEC_COST);
-  format %{ "vfmv.s.f $tmp, $src1_dst\t#@reduce_addD\n\t"
-            "vfredosum.vs $tmp, $src2, $tmp\n\t"
-            "vfmv.f.s $src1_dst, $tmp" %}
+  format %{ "reduce_addD $src1_dst, $src2\t# KILL $tmp" %}
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/riscv_v.ad:1393]: Similar here.

**Review by RealFYang — CHANGES_REQUESTED** [review]

File: src/hotspot/cpu/riscv/riscv_v.ad (hunk 2 above)
Comment (zifeihan) [reply src/hotspot/cpu/riscv/riscv_v.ad]: Thanks for the review, and has been fixed.

**Review by RealFYang — APPROVED** [review]

**Comment by zifeihan** [comment]
@RealFYang @yhzhu20 Thanks for the review.
/integrate

**Pull request closed without merging**